Fix KW issue

yuanzhel opened the pull request:
Add a if statement to check if fstat() return an error value.

sysopenci reviewed (commented):
Autobuild started from pull-request-changes on this PR.

FAILURE: CheckBug Bad comments/Bugs

For more information, see: /absp/builders/celadon_pmr0-autobuild/builds/106

yuanzhel committed:
Add some macros to support MDAPI

This change is related to MDAPI rebase used for OAM-91163.

Tracked-On: OAM-91163
Signed-Off-By: Liu, Yuanzhe <yuanzhe.liu@intel.com>
